src/main/java/org/launchcode/liftoffrecipetracker/models/Recipe.java

```diff
@@ -0,0 +1,107 @@
+package org.launchcode.liftoffrecipetracker.models;
+
+import javax.persistence.Entity;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.HashMap;
+
+
+@Entity
+public class Recipe extends AbstractEntity {
+	//properties
+	private HashMap<String, String> ingredients;
+
+	@NotBlank(message = "Recipe name required.")
+	@NotNull
+	@Size(min=2,max=50, message = "Recipe name must be between 2 and 50 characters long.")
+	private String name;
+
+	@NotNull
+	@NotBlank
+	@Size(min=4, max=1000, message = "Directions are required.")
+	private String directions;
+
+	@NotBlank
+	@Min(value = 1, message = "Must be at least one serving.")
+	private int servings;
+
+	private int cookTime;
+	private int prepTime;
+	private String image;
+
+	//constructors
+	public Recipe(HashMap<String, String> ingredients,
+	              @NotBlank(message = "Recipe name required.") @NotNull @Size(min = 2, max = 50, message = "Recipe name must be between 2 and 50 characters long.") String name,
+	              @NotNull @NotBlank @Size(min = 4, max = 1000, message = "Directions are required.") String directions,
+	              @NotBlank @Min(value = 1, message = "Must be at least one serving.") int servings) {
```

Comment:
I think you can remove the validation from the constructor. It is redundant here I believe and makes it more difficult to see what is actually required in the constructor. I think by including the @ Valid annotation in our controllers it forces things to be created properly.